docs/design_proposals/transparent-init.md

```diff
@@ -0,0 +1,94 @@
+# Transparent Init
+
+* Author(s): Marlon Gamez
+* Design Shepherd: \<skaffold-core-team-member\>
+* Date: 10/13/2020
+* Status: [Reviewed/Cancelled/Under implementation/Complete]
+
+## Background
+
+This proposal is brought about by an older issue, [#1273](https://github.com/GoogleContainerTools/skaffold/issues/1273)
+
+With the recent focus on skaffold UX, we've decided to see how we can improve the onboarding experience when using skaffold.
+One way we'd like to do this is by allowing the user to simply install skaffold and run any of the skaffold commands in their project, without having to first run `skaffold init` and go through the interactive portion of creating a config for their project.
+To do this, we'd like to automatically create a config upon invocation of skaffold commands, so that the user doesn't have to see something like this
+```
+❯ skaffold dev
+skaffold config file skaffold.yaml not found - check your current working directory, or try running `skaffold init`
+```
+when trying to run skaffold.
+
+## Design
+
+With transparent init, a user's first run of `skaffold dev` could look something like this:
+
+Given a simple project structure:
+```
+project-dir
+├── Dockerfile
+├── README.md
+├── k8s-pod.yaml
+└── main.go
+```
+Running skaffold dev:
+```
+❯ skaffold dev
+Skaffold config file not found. Generating temporary config for use this run.
+Listing files to watch...
+ - skaffold-example
+Generating tags...
+ - skaffold-example -> skaffold-example:v1.14.0-59-g7e5ae4cbc-dirty
+Checking cache...
+ - skaffold-example: Not found. Building
+Found [minikube] context, using local docker daemon.
+Building [skaffold-example]...
+Sending build context to Docker daemon  3.072kB
+...
+```
+
+There are two approaches to implementation that I've considered:
+
+**Creating a temporary `skaffold.yaml` by running `skaffold init --force`**
+
+Under the hood view:
+- `skaffold dev` is run
+- skaffold searches for a `skaffold.yaml` file in the directory, finds nothing
+- skaffold creates a new config using `skaffold init --force`
+- the new config is parsed and brought into memory
+- the config file is deleted from disk
+- `skaffold dev` continues like normal
+
+**Creating the config in memory using init functions**
+
+Under the hood view:
+- `skaffold dev` is run
```

Comment:
I like this approach. 

You can also add a flow where when user hit Cntrl-C, you present an option to the user if they would want to save the skaffold config ?

Reply:
I think this would be nice, maybe set some value that is then checked at the end of the run to prompt the user about this?

Reply:
I feel like we should just write it out automatically? it's not a big deal for the user to remove the file if they want to generate it interactively/write their own. it also gives them a starting point to start tweaking their config.

Reply:
agreed 👍🏼